src/providers/twitch/TwitchMessageBuilder.cpp

```diff
@@ -216,44 +216,41 @@ MessagePtr TwitchMessageBuilder::build()
 
                 // Actually just text
                 QString linkString = this->matchLink(string);
-                auto fontStyle = FontStyle::ChatMedium;
-
-                if (string.startsWith('@') && app->settings->enableUsernameBold) {
-                    fontStyle = FontStyle::ChatMediumBold;
-                }
 
                 Link link;
 
                 if (linkString.isEmpty()) {
                     link = Link();
+                    if (string.startsWith('@')) {
+                        this->emplace<TextElement>(string, TextElement::BoldUsername, textColor,
+                                                   FontStyle::ChatMediumBold);
+                        this->emplace<TextElement>(string, TextElement::NonBoldUsername, textColor);
+                    } else {
+                        this->emplace<TextElement>(string, TextElement::Text, textColor);
+                    }
                 } else {
-                    if (app->settings->lowercaseLink) {
-                        QRegularExpression httpRegex("\\bhttps?://",
-                                                     QRegularExpression::CaseInsensitiveOption);
-                        QRegularExpression ftpRegex("\\bftps?://",
-                                                    QRegularExpression::CaseInsensitiveOption);
-                        QRegularExpression getDomain("\\/\\/([^\\/]*)");
-                        QString tempString = string;
-
-                        if (!string.contains(httpRegex)) {
-                            if (!string.contains(ftpRegex)) {
-                                tempString.insert(0, "http://");
-                            }
+                    QRegularExpression httpRegex("\\bhttps?://",
+                                                 QRegularExpression::CaseInsensitiveOption);
+                    QRegularExpression ftpRegex("\\bftps?://",
+                                                QRegularExpression::CaseInsensitiveOption);
+                    QRegularExpression getDomain("\\/\\/([^\\/]*)");
+                    QString tempString = string;
+
+                    if (!string.contains(httpRegex)) {
+                        if (!string.contains(ftpRegex)) {
+                            tempString.insert(0, "http://");
                         }
-                        QString domain = getDomain.match(tempString).captured(1);
-                        string.replace(domain, domain.toLower());
                     }
+                    QString domain = getDomain.match(tempString).captured(1);
+                    QString lowercaseLinkString = string;
+                    lowercaseLinkString.replace(domain, domain.toLower());
```

Comment:
Replace only the domain.

In the example http://GoogLe.com/?redirect=GoogLe.com/asdf it would replace the second one as well

Reply:
```
const QString tempstring1 =
        "^"
        // Identifier for http and ftp
        "(?:(?:https?|ftps?)://)?"
        // user:pass authentication
        "(?:\\S+(?::\\S*)?@)?"
        "(?:"
        // IP address dotted notation octets
        // excludes loopback network 0.0.0.0
        // excludes reserved space >= 224.0.0.0
        // excludes network & broacast addresses
        // (first & last IP address of each class)
        "(?:[1-9]\\d?|1\\d\\d|2[01]\\d|22[0-3])"
        "(?:\\.(?:1?\\d{1,2}|2[0-4]\\d|25[0-5])){2}"
        "(?:\\.(?:[1-9]\\d?|1\\d\\d|2[0-4]\\d|25[0-4]))"
        "|"
        // host name
        "(?:(?:[_a-z\\x{00a1}-\\x{ffff}0-9]-*)*[a-z\\x{00a1}-\\x{ffff}0-9]+)"
        // domain name
        "(?:\\.(?:[a-z\\x{00a1}-\\x{ffff}0-9]-*)*[a-z\\x{00a1}-\\x{ffff}0-9]+)*"
        // TLD identifier
        "(?:\\.(?:[a-z\\x{00a1}-\\x{ffff}]{2,}))"
        "\\.?"
        ")"
        // port number
        "(?::\\d{2,5})?";
QRegularExpression getDomain(tempstring1, QRegularExpression::CaseInsensitiveOption);
QString tempString = string;
QString domain = getDomain.match(tempString).captured(0);
QString lowercaseLinkString = string;
lowercaseLinkString.replace(domain, domain.toLower());
```